### PR TITLE
Run2 Process Xsecs

### PR DESCRIPTION
**Pull request by Bogdan-Wiederspan**
This PR adds all xsecs for Run2

### CODE REVIEW

**Comment by riga** [comment]
@Bogdan-Wiederspan What would need to be done for this PR to leave the WIP phase?

**Comment by Bogdan-Wiederspan** [comment]
Nothing needs to be added @riga. I simply forgot to switch the state to "ready to review"

**Comment by nprouvost** [comment]
Actually not completely true, I am still including updated values using the GenXSecAnalyzer directly instead of cmsdb and there will be a few new processes to include once the run2 campaigns are merged (e.g. several multibosons decay channels like ZZ->QQ)

**Comment by nprouvost** [comment]
Updated values are now fully included with the corresponding logs and some logs expected to be needed after the merge, only the additional values which will come with the processes added with the merge are missing.

**Comment by haddadanas** [comment]
resolved Merge conflicts with master :)
Question: If we have a $\pm 2$% uncert is it noted in the scinum version we use as `uncert=2j` or `uncert=0.02j`? A few xsecs use the first notation
I tested it in the latest version of scinum and the latter would be correct, but I wasn't sure if the notation was different in older scinum versions.
I made a commit to correct these and can push it if the second notation turns out to be correct :sweat_smile:

**Comment by nprouvost** [comment]
Nice, thanks for spotting these

File: cmsdb/campaigns/run2_2016_HIPM_nano_uhh_v12/ewk.py
Comment (mafrahm) [inline cmsdb/campaigns/run2_2016_HIPM_nano_uhh_v12/ewk.py:92]: I think it is a good idea to have all dataset names as all lower case.

However, changing the name of processes will impact all ongoing analyses because users will need to rename all of their datasets + all outputs that were produced using this dataset. Therefore, I have two comments to add:

- How can we implement such a change without impacting ongoing analyses too much? Can we provide some script that automatically changes the path of all outputs that have been produced with one of the affected datasets?
- When implementing this convention, we should require this for all datasets. Are these all datasets with uppercase letters or are there still datasets/processes remaining with uppercase letters? Maybe we should even add this requirement to our tests?

File: cmsdb/processes/ewk.py
Comment (mafrahm) [inline cmsdb/processes/ewk.py:402]: The previous cross sections were provided from the xsdb: https://xsdb-temp.app.cern.ch
Is there a reason/advantage to use the cross sections from the GenXSecAnalyzer?

Comment (nprouvost) [reply cmsdb/processes/ewk.py:402]: It is the same method of calculation as the entries on xsdb (just getting the xsec value used for generating the miniaods thanks to the tool), only that 1) we know exactly what parameters were used for the calculation (e.g. how many events we average on, which cmssw version was used...), 2) we are getting the corresponding logs and 3) we are probably more precise as we are using 5M events when possible which seems to give much more precise results than what is seen in xsdb (probably mostly done with the default number of events: 1M)

**Comment by riga** [comment]
> Question (_resolved_): If we have a % uncert is it noted in the scinum version we use as `uncert=2j` or `uncert=0.02j`? A few xsecs use the first notation

`0.02j` is indeed the right way to do it 👍

**Comment by riga** [comment]
@nprouvost @mafrahm Are the two comments above resolved?

File: cmsdb/campaigns/run2_2016_HIPM_nano_uhh_v12/ewk.py
Comment (riga) [reply cmsdb/campaigns/run2_2016_HIPM_nano_uhh_v12/ewk.py:92]: Not yet. I added an aux flag `"allow_uppercase_name": True` to the ggHH processes that bypasses the lower case test. Since these are mostly bbww processes and datasets, I'd leave this to you @mafrahm. So feel free to rename them at some point and remove the flags again.

Comment (riga) [reply cmsdb/campaigns/run2_2016_HIPM_nano_uhh_v12/ewk.py:92]: Btw, I noticed a lot of inconsistencies in the names of processes / datasets in cases of open intervals on, for instance, `pt` or `mgg`. When there is just a single number in the dataset, `pT30`, this usually translates into `pt30toinf`. We should talk about this in the next meeting.

**Review by riga — APPROVED** [review]

**Pull request merged**
